core/auditlog/auditlog.go

```diff
@@ -245,13 +248,37 @@ func idString(id uint64) string {
 // primeLogFile ensures the logsink log file is created with the
 // correct mode and ownership.
 func primeLogFile(path string) error {
-	f, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY, 0600)
+	permissions := os.FileMode(0640)
+	f, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY, permissions)
 	if err != nil {
 		return errors.Trace(err)
 	}
 	if err := f.Close(); err != nil {
 		return errors.Trace(err)
 	}
-	err = utils.ChownPath(path, "syslog")
-	return errors.Trace(err)
+	wantedOwner, wantedGroup := systemd.SyslogUserGroup()
+	return SetOwnerGroupLogPermissions(path, wantedOwner, wantedGroup, permissions)
+}
+
+func SetOwnerGroupLogPermissions(filePath string, wantedOwner string, wantedGroup string, permissions os.FileMode) error {
```

Comment:
This function then should live in the `core/paths` package. And I don't think it should deal with `Chown`.

Reply:
I tried to incorporate your feedback. Most makes a lot of sense!
But I don't know if I agree with this one, I think the "SetOwnerGroupLog" should do a chown, right? It is the main and sole purpose of this function.